manifests/check.pp

```diff
@@ -236,10 +238,53 @@
     publish             => $publish,
     dependencies        => $dependencies,
     custom              => $custom,
```

Comment:
That's effectively what sensu_check_content() does, with content and custom.  It could definitely be a little cleaner.  I used a function because my first pass at manipulating immutable Hashes the Puppet DSL turned out awkward.  Is there a better way to handle the behavior of content and custom to get the final JSON?

Something like:

```
# as before, with name, custom and content stripped out.
$resource_hash = { ... }
# Final structure of the output.
$content_hash = $content + {'checks' => { "$check_name" => $resource_hash + $custom } }
```

The only other thing is to deal with all of the undefined and `"absent"` values, so this should work:

```puppet

$resource_hash_real = $resource_hash.reduce({}) |$memo, $kv| {
  if ($kv[1] == "absent" or $kv[1] == undef) {
    $memo
  } else {
    $memo + Hash.new($kv)
  } 
}
```

I'll replace the function with this Puppet DSL and some comments describing the structure.